rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectValidator.java

```diff
@@ -0,0 +1,33 @@
+/*Copyright Siemens AG, 2019. Part of the SW360 Portal Project.
+
+ SPDX-License-Identifier: EPL-1.0
+
+ All rights reserved. This program and the accompanying materials
+ are made available under the terms of the Eclipse Public License v1.0
+ which accompanies this distribution, and is available at
+ http://www.eclipse.org/legal/epl-v10.html
+*/
+package org.eclipse.sw360.rest.resourceserver.project;
+
+import java.util.Set;
+
+import org.eclipse.sw360.datahandler.thrift.projects.Project;
+import org.eclipse.sw360.rest.resourceserver.Sw360ResourceServer;
+import org.springframework.stereotype.Component;
+
+/*
+ * This class is used to validate the project input fields.
+ * @author nutan.das@siemens.com
+ * */
+@Component
+public class ProjectValidator {
```

Comment:
I like it a lot that you started to use input validation! Do you see any possibility to achieve the same by using more standardized techniques (based on JSRs) like mentioned in https://www.mkyong.com/spring-boot/spring-rest-validation-example/, https://www.baeldung.com/spring-boot-bean-validation or https://www.baeldung.com/spring-data-rest-validators?

Reply:
Hi @imaykay ,

Currently JSR validation is not working for @BasePathAwareController.(
https://github.com/spring-projects/spring-boot/issues/12863
https://jira.spring.io/browse/DATAREST-593
).
And I don't see any better way to do the validation here.Please suggest if there is any better solution to it.